prusa/link/web/main.py

```diff
@@ -428,23 +428,42 @@ def api_update(req, env):
     versions"""
     # pylint: disable=unused-argument
     headers = {"Update-Available": "False"}
-    updatable = []
 
     if env == "python":
-        output = check_output([executable, '-m', 'pip', 'list', '-o']).decode()
-        packages_list = output.split('\n')[2:-1]
-        if packages_list:
-            headers["Update-Available"] = "True"
-        for package_ in packages_list:
-            package = package_.split()
-            updatable.append(
-                {
-                    "name": package[0],
-                    "version": package[1],
-                    "available_version": package[2],
-                    "path": None}
-            )
-        return JSONResponse(available_updates=updatable, headers=headers)
+        try:
+            output = check_output(
+                [executable, '-m', 'pip', 'install', '--dry-run', '-U',
+                 'prusalink']).decode()
+
+        # There's a problem with package installation, or it does not exist
+        except CalledProcessError as exception:
+            raise conditions.UnavailablePackage() from exception
```

Comment:
Error must be different, like UnavailableUpdate.... may be with some additional information, for for example, what failed.

And, if i read right, you catch only check_output error, bot not pip error right?